application/src/main/java/org/thingsboard/server/service/transport/DefaultTransportApiService.java

```diff
@@ -281,6 +283,13 @@ private ListenableFuture<TransportApiResponseMsg> handle(GetOrCreateDeviceFromGa
             Lock deviceCreationLock = deviceCreationLocks.computeIfAbsent(requestMsg.getDeviceName(), id -> new ReentrantLock());
             deviceCreationLock.lock();
             try {
+                DeviceProfile deviceProfile = deviceProfileCache.findOrCreateDeviceProfile(gateway.getTenantId(), requestMsg.getDeviceType());
```

Comment:
This code is not needed here at all. It should be reverted entirely. We should receive information about isSparkplug from the GetOrCreateDeviceFromGatewayRequestMsg. I will do the change